### PR TITLE
ensure value/esdt/nft is transferred back always

### PR DESCRIPTION
**Pull request by sasurobert**
ensure value/esdt/nft is transferred back always

### CODE REVIEW

File: arwen/host/execution.go [hunk 1]
```diff
@@ -691,16 +693,20 @@ func (host *vmHost) RevertESDTTransfer(input *vmcommon.ContractCallInput) {
...
 		log.Error("RevertESDTTransfer failed", "returnCode", vmOutput.ReturnCode, "returnMessage", vmOutput.ReturnMessage)
+		host.meteringContext.UseGas(host.meteringContext.GasLeft())
+		host.runtimeContext.FailExecution(errors.New(vmOutput.ReturnMessage))
 	}
 }
 
 // ExecuteESDTTransfer calls the process built in function with the given transfer for ESDT/ESDTNFT if nonce > 0
 // there are no NFTs with nonce == 0
-func (host *vmHost) ExecuteESDTTransfer(destination []byte, sender []byte, tokenIdentifier []byte, nonce uint64, value *big.Int, callType vmcommon.CallType) (*vmcommon.VMOutput, uint64, error) {
+func (host *vmHost) ExecuteESDTTransfer(destination []byte, sender []byte, tokenIdentifier []byte, nonce uint64, value *big.Int, callType vmcommon.CallType, isRevert bool) (*vmcommon.VMOutput, uint64, error) {
```
Comment (iulianpascalau) [inline arwen/host/execution.go:709]: Instead of using a bool variable, we could have had 2 functions for this.

File: arwen/host/execution.go [hunk 2]
```diff
@@ -14,6 +14,8 @@ import (
 	"github.com/ElrondNetwork/elrond-go/core/vmcommon"
 )
 
+const gasLimitToUseForRevertESDT = uint64(1_500_000_000)
```
Comment (iulianpascalau) [inline arwen/host/execution.go:17]: why hardcoding this here?

**Review by camilbancioiu — APPROVED** [review]

**Review by andrei-marinica — APPROVED** [review]

File: arwen/elrondapi/elrondei.go [hunk 3]
```diff
@@ -867,6 +867,10 @@ func transferValueExecute(
 		_, _, _, err = host.ExecuteOnDestContext(contractCallInput)
 		if err != nil {
 			logEEI.Trace("eGLD pre-transfer execution failed", "error", err)
+			err = output.TransferValueOnly(send, dest, big.NewInt(0).SetBytes(valueBytes), false)
```
Comment (iulianpascalau) [inline arwen/elrondapi/elrondei.go:870]: Is this correct?

**Review by iulianpascalau — CHANGES_REQUESTED** [review]
For above comments

File: arwen/host/execution.go (hunk 2 above)
Comment (sasurobert) [reply arwen/host/execution.go:17]: where should I hardcode it ?

File: arwen/elrondapi/elrondei.go (hunk 3 above)
Comment (sasurobert) [reply arwen/elrondapi/elrondei.go:870]: yes. this is not needed.

File: arwen/host/execution.go (hunk 1 above)
Comment (sasurobert) [reply arwen/host/execution.go:709]: 2 functions with copy pasted code ?

Comment (iulianpascalau) [reply arwen/host/execution.go:709]: 2 functions with extracted duplicated code

File: arwen/host/execution.go (hunk 2 above)
Comment (iulianpascalau) [reply arwen/host/execution.go:17]: do we pass to arwen the maximum gas limit per miniblock?

Comment (iulianpascalau) [reply arwen/host/execution.go:17]: I see that in host.go we have some consts defined

File: arwen/elrondapi/elrondei.go (hunk 3 above)
Comment (iulianpascalau) [reply arwen/elrondapi/elrondei.go]: ok

File: arwen/host/execution.go (hunk 1 above)
Comment (iulianpascalau) [reply arwen/host/execution.go:709]: can leave as it is for now

**Review by iulianpascalau — APPROVED** [review]

**Review by andrei-marinica — APPROVED** [review]

**Pull request merged**